ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandRunner.java

```diff
@@ -16,7 +16,9 @@
 package io.confluent.ksql.rest.server.computation;
 
 import com.google.common.annotations.VisibleForTesting;
+import io.confluent.ksql.rest.Errors;
```

Comment:
I think all our state tracking logic in this class can be simplified. The current pattern is to set a bunch of booleans and compute status from every request. This logic is getting pretty complex, and we're polluting the interface here with a bunch of status-getting methods (e.g. getStatus, getDegradedReason, getDegradedMsg, etc). I think it would be simpler to just have the command runner explicitly set the status:

```
public static enum CommandRunnerDegradedReason {
    Function<Errors, String> msgFactory;

    public String getMsg(final Errors errors) {
        return msgFactory(errors);
    }

    CommandRunnerDegradedReason(Function<Errors, String> msgFactory) {
        this.msgFactory = msgFactory;
    }

    NONE(errors -> ""),
    CORRUPTED(Errors:: commandRunnerDegradedBackupCorruptedErrorMessage),
    INCOMPATIBLE_COMMAND(Errrors:: commandRunnerDegradedIncompatibleCommandsErrorMessage)
}

public static class Status {
    CommandRunnerStatus state;
    Optional<CommandRunnerDegradedReason> degradedReason;

    public CommandRunnerStatus getState() {
        return state;
    }
    public CommandRunnerDegradedReason getDegradedReason() {
        return degradedReason;
    }
}

private Status state = new Status(RUNNING, Optional.empty());

public Status getState() {
    return State; 
} 

public void run() {
    try {
        while (!closed) {
            if (incompatibleCommandDetected) {
                state = new Status(DEGRADED, Optional.of(INCOMPATIBLE_COMMAND))
                closeEarly(); 
            }  else if (backup.corrupted()) {
                state = new Status(DEGRADED, Optional.of(CORRUPTED))
                closeEarly(); 
            } else {
                 …
…

```

Reply:
Done